cassandra/components.py

```diff
@@ -712,86 +714,259 @@ def prep_for_xanthos(self, monthly_data):
+    pr[0], tas[1] with pr[1], and so on.  Mixing the tas and pr grids from two
+    different realizations (e.g., tas[0] wth pr[2]) is not valid and should be
+    avoided.
 
-    results: pr and temp grids
     """
 
     def __init__(self, cap_tbl):
         super(FldgenComponent, self).__init__(cap_tbl)
         self.addcapability("gridded_pr")
         self.addcapability("gridded_tas")
+        self.addcapability('gridded_pr_coord')
+        self.addcapability('gridded_tas_coord')
 
+        
+    def finalize_parsing(self):
+        self.params['loadpkgs'] = util.parseTFstring(self.params['loadpkgs'])
+        self.params['ngrids'] = int(self.params['ngrids'])
+        self.params['startyr'] = int(self.params['startyr'])
+        self.params['nyear'] = int(self.params['nyear'])
+        
+        
     def run_component(self):
         """Run the fldgen and an2month R scripts."""
         from rpy2.robjects.packages import importr
         import rpy2.robjects as robjects
         import numpy as np
+        from rpy2.robjects import numpy2ri
+        numpy2ri.activate() # enable automatic conversion of numpy objects to R equivalents.
 
-        workdir = self.params["workdir"]
-
-        an2month = os.path.join(workdir, "an2month")
-        fldgen = os.path.join(workdir, "fldgen")
-        gexp_funs = os.path.join(workdir, "frontEnd_grandExp/grand_experiment_functions.R")
-        emulator_mapping = os.path.join(workdir, "input", "emulator_mapping.csv")
-        hector_runs = os.path.join(workdir, "input", "hector_tgav-rcp45.csv")
-
-        devtools = importr("devtools")
-        devtools.load_all(an2month)
-        devtools.load_all(fldgen)
-
-        rsource = robjects.r["source"]
-        rsource(gexp_funs)
-
-        grand_experiment = robjects.r["grand_experiment"]
-        r = grand_experiment(mapping=emulator_mapping, tgavSCN=hector_runs, xanthos_dir='', N=1)
-
-        # r is a complex nested structure:
-        # Emulator1 (e.g. ipsl-cm5a-lr)
-        # -- Scenario1 (e.g. rcp45)
-        # -- -- RunNum1 (e.g. 1)
-        # -- -- -- Monthly Precipitation
-        # -- -- -- -- Precip Data (months x 67420)
-        # -- -- -- -- Precip Coordinates (3 (cell, lon, lat) x 67420)
-        # -- -- -- -- Precip Units (e.g. mm_month-1)
-        # -- -- -- Monthly Temperature
-        # -- -- -- -- Temp Data (months x 67420)
-        # -- -- -- -- Temp Coordinates (3 (cell, lon, lat) x 67420)
-        # -- -- -- -- Temp Units (e.g. C)
-        # -- -- RunNum2
-        # -- -- -- ...
-        # -- Scenario2
-        # -- -- ...
-        # Emulator2
-        # -- ...
-        # ...
-        emulator_name = r[0]
-        run_name = emulator_name[0]
-        run_N = run_name[0]
-
-        fldgen_pr = {
-            'data': np.asarray(run_N[0][0]),
-            'coords': np.asarray(run_N[0][1]),
-            'units': run_N[0][2][0]  # All R strings are in vectors; take 1st element
-        }
+        if self.params['loadpkgs']:
```

Comment:
For explicit loading, we will still need `workdir` to be defined. I'm thinking, though, that maybe we should not support   this and require that the packages be already installed, as we do with Xanthos.

Reply:
That's correct.  I changed `workdir` to `pkgdir`, to better reflect its function.

Reply:
As to the question of supporting or not supporting this usage, I can see an argument for either.  If you don't have strong feelings either way, then I'd say that since we have the code for on-demand loading already in, let's just leave it.  Maybe we'll call it "deprecated" at some point.

Reply:
Good with me!